core/templates/services/keyboard-shortcut.service.spec.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the keyboard shortcut service.
+ */
+import 'mousetrap';
+
+import { KeyboardShortcutService } from 'services/keyboard-shortcut.service';
+
+describe('KeyboardShortcutService', () => {
+  jasmine.getEnv().allowRespy(true);
+  const keyboardShortcutService = new KeyboardShortcutService();
+
+  it('should test navigation shortcuts.', () => {
```

Comment:
I am testing whether the keyboard shortcuts can successfully be bound and can properly be triggered. Is that too limited? I mostly do the verification in the E2E testing. I guess I'm a little confused about the purpose of unit tests vs E2E tests.

Reply:
I see. Then, how about we use the spy on the `setHref` function to record which URLs have been set after calling the function. You can then at least assert that `setHref` has been called for the expected URLs. Otherwise, this test isn't really doing anything.